### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by nguuuquaaa** [VariantXYZ/PSO2DamageDump#1]
So I tried compiling the DLL (without changing anything) with Visual Studio 2017 (and tons of dependencies). Everything went well, compiler wouldn't complain anything, until when I put it into actual use, the game crashes on character spawn.

I tried to print out stuff to diagnose the problem and it seems pkt->dataSize returns a mysteriously big number, so malloc runs wild and crashes everything.

I seriously don't understand what's going on. Is it VS fault? Or pso2h.dll breaking?
Either way can I get some direction on solving this issue?

**Comment by VariantXYZ** [comment]
It's probably that I need to update the pso2h binaries (the actual packet is passed through pso2h), though I don't think I've updated it in like a year...

**Comment by nguuuquaaa** [comment]
The thing is, the compiled dll that comes with Overparse (with dates back to 2016) is still usable. So now I'm questioning my VS compiling ability.

**Comment by VariantXYZ** [comment]
Do you crash using the pso2h.dll inside the pso2h binaries folder? I'm pretty sure I probably changed the API from the one that's packaged with Overparse.

**Comment by nguuuquaaa** [comment]
Yep, crash with either the one in pso2h binaries or the one that comes with tweaker.

**Comment by VariantXYZ** [comment]
I'll have to investigate myself, but I won't be able to til next week. I'm guessing you're hitting the issue in 'void getNames(LPBYTE packet)' first, since it's the character spawn packet.

Try printing out the first 8 bytes of the packet with something like:
`
for(int i = 0; i < 8; i++) PSO2Hook::pso2hLogLine("%02X ", packet[i]);`

**Comment by nguuuquaaa** [comment]
This appeared in the log file
```
[1523669148][PSO2DamageDump] 30 
[1523669148][PSO2DamageDump] 1E 
[1523669148][PSO2DamageDump] 3A 
[1523669148][PSO2DamageDump] 1A 
[1523669148][PSO2DamageDump] C6 
[1523669148][PSO2DamageDump] 63 
[1523669148][PSO2DamageDump] 59 
[1523669148][PSO2DamageDump] 04 
```

**Comment by etherfactor** [comment]
By any chance, were you able to figure out what was causing the crash?
